app/share.js

```diff
@@ -14,61 +14,114 @@ import {
 } from './utils/theme';
 import Navigation from './lib/ShareNavigation';
 import store from './lib/createStore';
-import sharedStyles from './views/Styles';
-import { hasNotch, supportSystemTheme } from './utils/deviceInfo';
-import { defaultHeader, onNavigationStateChange, cardStyle } from './utils/navigation';
+import { supportSystemTheme } from './utils/deviceInfo';
+import {
+	defaultHeader, themedHeader, getActiveRouteName, navigationTheme
+} from './utils/navigation';
 import RocketChat, { THEME_PREFERENCES_KEY } from './lib/rocketchat';
 import { ThemeContext } from './theme';
 import { localAuthenticate } from './utils/localAuthentication';
 import ScreenLockedView from './views/ScreenLockedView';
 
-const InsideNavigator = createStackNavigator({
-	ShareListView: {
-		getScreen: () => require('./views/ShareListView').default
-	},
-	ShareView: {
-		getScreen: () => require('./views/ShareView').default
-	},
-	SelectServerView: {
-		getScreen: () => require('./views/SelectServerView').default
-	}
-}, {
-	initialRouteName: 'ShareListView',
-	defaultNavigationOptions: defaultHeader,
-	cardStyle
-});
-
-const OutsideNavigator = createStackNavigator({
-	WithoutServersView: {
-		getScreen: () => require('./views/WithoutServersView').default
-	}
-}, {
-	initialRouteName: 'WithoutServersView',
-	defaultNavigationOptions: defaultHeader,
-	cardStyle
-});
-
-const AppContainer = createAppContainer(createSwitchNavigator({
-	OutsideStack: OutsideNavigator,
-	InsideStack: InsideNavigator,
-	AuthLoading: {
-		getScreen: () => require('./views/AuthLoadingView').default
-	}
-},
-{
-	initialRouteName: 'AuthLoading'
-}));
+// Outside Stack
+import WithoutServersView from './views/WithoutServersView';
+
+// Inside Stack
+import ShareListView from './views/ShareListView';
+import ShareView from './views/ShareView';
+import SelectServerView from './views/SelectServerView';
+import { setCurrentScreen } from './utils/log';
+import AuthLoadingView from './views/AuthLoadingView';
+
+const Inside = createStackNavigator();
+const InsideStack = () => {
+	const { theme } = useContext(ThemeContext);
+
+	const screenOptions = {
+		...defaultHeader,
+		...themedHeader(theme)
+	};
+	screenOptions.headerStyle = {
+		...screenOptions.headerStyle,
+		// TODO: fix on multiple files PR :)
+		height: 57
```

Comment:
I think shareListView should have this same value at Header
![WhatsApp Image 2020-06-12 at 17 00 33](https://user-images.githubusercontent.com/29778115/84541748-5d7dfe00-acce-11ea-9882-87bb16acb83e.jpeg)
Look below the header, a weird margin.

Reply:
It's going to be fixed on multiple files PR